src/main/java/core/basesyntax/service/Figure.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax.service;
+
+import core.basesyntax.supplier.ColorSupplier;
+import core.basesyntax.supplier.FigureSupplier;
+
+public abstract class Figure implements FigureBehaviourService {
+    protected ColorSupplier colorSupplier = new ColorSupplier();
+    private String color;
+
+    public Figure(String color) {
+        this.color = color;
+    }
+
+    public String getColor() {
+        return color;
+    }
+
+    public void setColor(String color) {
+        this.color = color;
+    }
+
+    @Override
+    public void drawFigure() {
+        FigureSupplier figureSupplier = new FigureSupplier(new ColorSupplier());
```

Comment:
`drawFigure` is meant to be a fundamental behavior that every subclass must implement in their own way, it should be declared as an abstract method in the Figure class. This ensures that all concrete subclasses must provide their specific implementation.

so, I would suggest removing the implementation of drawFigure here.